test/burn/org/apache/cassandra/net/MessageGenerator.java

```diff
@@ -159,15 +158,13 @@ static Header readHeader(DataInputPlus in, int messagingVersion) throws IOExcept
 
     static void writeLength(byte[] payload, DataOutputPlus out, int messagingVersion) throws IOException
     {
-        if (messagingVersion < VERSION_40)
-            out.writeInt(payload.length);
-        else
-            out.writeUnsignedVInt32(payload.length);
+        assert messagingVersion >= VERSION_40;
+        out.writeUnsignedVInt32(payload.length);
     }
 
     static long serializedSize(byte[] payload, int messagingVersion)
     {
-        return payload.length + (messagingVersion < VERSION_40 ? 4 : VIntCoding.computeUnsignedVIntSize(payload.length));
+        return payload.length + VIntCoding.computeUnsignedVIntSize(payload.length);
```

Comment:
`serializedSize` doesn't use the `messagingVersion` parameter anymore, we can remove it.